.github/workflows/go-mod-auto-bump.yml

```diff
@@ -0,0 +1,40 @@
+# This workflow is used in order to automatically update go.mod in case one of:
+# * osmoutils
+# * osmomath
+# * ibc-hooks
+# is updated when a push to the main branch happened (including merged PRs)
```

Comment:
I saw previous discussions but I still think that we need to figure out a way to commit directly to the developer's branch / PR.

If you make a breaking change to `osmoutils`, `osmomath`, or `x/ibc-hooks`, e2e CI fails unless you manually retag.

The biggest value add of this feature IMO is automating such manual retagging. Please let me know what you think